phylanx/util/detail/range_dimension.hpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2020 Bita Hasheminezhad
+// Copyright (c) 2020 Hartmut Kaiser
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(PHYLANX_UTIL_DETAIL_RANGE_DIMENSION)
+#define PHYLANX_UTIL_DETAIL_RANGE_DIMENSION
+
+#include <array>
+#include <cstddef>
+#include <string>
+
+namespace phylanx { namespace util { namespace detail {
+
+    inline std::size_t extract_range_num_dimensions(
+        phylanx::ir::range const& shape)
+    {
+        return shape.size();
+    }
+
+    inline std::array<std::size_t, PHYLANX_MAX_DIMENSIONS>
+    extract_positive_range_dimensions(phylanx::ir::range const& shape)
+    {
+        std::array<std::size_t, PHYLANX_MAX_DIMENSIONS> result = {0};
+        if (!shape.empty())
+        {
+            if (shape.size() == 1)
+            {
+                result[0] = extract_scalar_positive_integer_value_strict(
+                    *shape.begin());
+            }
```

Comment:
Could you add the name and filename to these `extract_` functions as well, please?

Reply:
Oh, sorry I missed that